src/main/java/org/sagebionetworks/web/client/widget/team/OpenUserInvitationsWidget.java

```diff
@@ -79,47 +81,67 @@ public void onFailure(Throwable caught) {
 	public void configure(String teamId, Callback teamRefreshCallback) {
 		this.teamId = teamId;
 		this.teamRefreshCallback = teamRefreshCallback;
-		profiles = new ArrayList<UserProfile>();
-		invitations = new ArrayList<MembershipInvtnSubmission>();
-		createdOnDates = new ArrayList<>();
 		currentOffset = 0;
 		getNextBatch();
-	};
+	}
 
 	@Override
 	public void getNextBatch() {
 		synAlert.clear();
-		//using the given team, try to show all pending membership requests (or nothing if empty)
+		// Show up to INVITATION_BATCH_LIMIT invitations extended by teamId
 		synapseClient.getOpenTeamInvitations(teamId, INVITATION_BATCH_LIMIT, currentOffset, new AsyncCallback<ArrayList<OpenTeamInvitationBundle>>() {
 			@Override
 			public void onSuccess(ArrayList<OpenTeamInvitationBundle> result) {
 				currentOffset += result.size();
-				
-				//create the associated object list, and pass to the view to render
-				for (OpenTeamInvitationBundle b : result) {
-					invitations.add(b.getMembershipInvtnSubmission());
-					profiles.add(b.getUserProfile());
-					String createdOn = dateTimeUtils.convertDateToSmallString(b.getMembershipInvtnSubmission().getCreatedOn());
-					createdOnDates.add(createdOn);
-				}
-				view.configure(profiles, invitations, createdOnDates);
-				
-				//show the more button if we maxed out the return results
-				view.setMoreResultsVisible(result.size() == INVITATION_BATCH_LIMIT);
+				addInvitations(result);
+				updateMoreButton(result.size());
 				gwt.restoreWindowPosition();
 			}
 			
 			@Override
 			public void onFailure(Throwable caught) {
 				if (caught instanceof NotFoundException) {
-					view.setMoreResultsVisible(false);
+					view.hideMoreButton();
 				} else {
 					synAlert.handleException(caught);
 				} 
 			}
 		});
 	}
-	
+
+	private void addInvitations(List<OpenTeamInvitationBundle> bundles) {
+		view.hideInvitations();
```

Comment:
Does this mean that all invitations hidden while these new invitation widgets are created, and then they are all shown again (if so, why?)?

Reply:
You're right, this hiding + showing is unnecessary.